gateway/ln-gateway/src/rpc/lnrpc_client.rs

```diff
@@ -0,0 +1,50 @@
+use std::{fmt::Debug, sync::Arc};
+
+use async_trait::async_trait;
+use fedimint_api::dyn_newtype_define;
+use tonic::Streaming;
+
+use crate::{
+    gatewaylnrpc::{
+        CompleteHtlcsRequest, CompleteHtlcsResponse, GetPubKeyResponse, PayInvoiceRequest,
+        PayInvoiceResponse, SubscribeInterceptHtlcsResponse,
+    },
+    Result,
+};
+
+/// Convenience wrapper around `GatewayLightningClient` protocol spec
+/// This provides ease in constructing rpc requests
+#[async_trait]
+pub trait ILnRpcClient: Debug + Send + Sync {
+    /// Get the public key of the lightning node
+    async fn get_pubkey(&self) -> Result<GetPubKeyResponse>;
+
+    /// Attempt to pay an invoice using the lightning node
+    async fn pay_invoice(
+        &self,
+        invoices: Vec<PayInvoiceRequest>,
+    ) -> Result<Streaming<PayInvoiceResponse>>;
```

Comment:
After #1426, only `subscribe_intercept_htlcs` exposes `tonic::Streaming` type.

Now, `Streaming` has a complex `Item = std::result::Result<SubscribeInterceptHtlcsResponse, Status>` where `Status` is yet another nuanced tonic type we'd need to mock redefine.

To make mocking of `ILnRpcClient` easy, I aliased the tonic type : `pub type HtlcStream = Streaming<SubscribeInterceptHtlcsResponse>;`. This way, testing libs don't have to explicitly take a dependency on tonic.

Mocking `Ok` results should be easy. Mocking `Err` results might still be challenging since response would need to be a tonic `Status`.

Reply:
#1636

Reply:
#1636